### DISCUSSION TITLE
to_unix_timestamp, get error "invalid timestamp format: input contains invalid characters"

### DISCUSSION THREAD

**Discussion by StefanSa** [vectordotdev/vector#10634] [Q&A] [answered]
hi there,
i have this here:
```
.event.start = join!([.date, .time], separator: " ")
.event.start = parse_timestamp!(.event.start, format: "%Y-%m-%d %H:%M:%S")
.nano_start = to_unix_timestamp(t'.event.start', unit: "milliseconds")
```
event.start a correct timestamp, `2021-12-30T15:59:34Z`but get this error message.
```
error[E601]: invalid timestamp
  ┌─ :6:33
  │
6 │ .nano_start = to_unix_timestamp(t'.event.start', unit: "milliseconds")
  │                                 ^^^^^^^^^^^^^^^ invalid timestamp format: input contains invalid characters
  │
  = see documentation about timestamps at https://vrl.dev/expressions/#timestamp
  = see language documentation at https://vrl.dev
```

**Comment by spencergilbert** [comment, marked as answer]
Hey @StefanSa I think you just need to remove the `t'...'` from that last line. That's instructing VRL to treat the contents as a _literal timestamp_, you just want the value at that path so just doing `to_unix_timestamp(.event.start, unit: "milliseconds")` should work for you.

**Comment by StefanSa** [comment]
@spencergilbert 
thanks for the tip, this now works as it should.
`to_unix_timestamp!(.event.start, unit: "milliseconds")`
